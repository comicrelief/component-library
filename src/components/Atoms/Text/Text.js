import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

/**
 * Text component
 */
export const BaseText = styled.span`
  color: ${({ color, theme }) =>
    color ? theme.color(color) : 'inherit'};
  font-size: ${({ size, theme }) => theme.fontSize(size)};
`;

/** Text renders different elements based on the `tag` prop */
const Text = ({ tag, children, ...rest }) => (
  <BaseText {...rest} as={tag}>
    {children}
  </BaseText>
);

Text.defaultProps = {
  tag: 'span',
  weight: 'normal',
  uppercase: false,
  size: 's0'
};

Text.propTypes = {
  /** Weight of Font */
  weight: PropTypes.string,
  /** Sets text transform to uppercase. */
  uppercase: PropTypes.bool,
  /** Colors */
  color: PropTypes.string,
  /** Tag type */
  tag: PropTypes.string,
  /** Sizes */
  size: PropTypes.string
};

export default Text;
