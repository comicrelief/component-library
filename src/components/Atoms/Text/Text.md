```js
<div>
  <Text tag="h3" size="s">
    Heading 3
  </Text>
  <Text tag="p" colour="black">
    My paragraph
  </Text>
  <Text tag="h3" size="m">
    Heading 3 size 3
  </Text>
  <Text tag="p" size="xl" color="red">
    My paragraph small and yellow
  </Text>
</div>
```

Sport Relief

```js
  <div>
    <Text tag="h3" size="s">
      Heading 3
    </Text>
    <Text tag="p" colour="black">
      My paragraph
    </Text>
    <Text tag="h3" size="m">
      Heading 3 size 3
    </Text>
    <Text tag="p" size="xl" color="red">
      My paragraph small and yellow
    </Text>
  </div>
```
