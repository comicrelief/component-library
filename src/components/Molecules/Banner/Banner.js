import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import spacing from '../../../theme/shared/spacing';
import { sizes } from '../../../theme/shared/allBreakpoints';

const Wrapper = styled.div`
  background: ${({ theme, backgroundColor }) => theme.color(backgroundColor)};
`;

const Container = styled.div`
  text-align: center;
  padding: calc(${spacing('m')} * 2) ${spacing('m')};
  margin: 0 auto;
  max-width: ${sizes.XL}px;

  @media ${({ theme }) => theme.allBreakpoints('M')} {
    padding: ${spacing('xl')} ${spacing('l')};
  }

  @media ${({ theme }) => theme.allBreakpoints('XL')} {
    padding: ${spacing('xl')} ${spacing('xl')};
  }
`;

const Banner = ({ children, backgroundColor }) => (
  <Wrapper backgroundColor={backgroundColor}>
    <Container>
      { children }
    </Container>
  </Wrapper>
);

Banner.propTypes = {
  children: PropTypes.node.isRequired,
  backgroundColor: PropTypes.string
};

Banner.defaultProps = {
  backgroundColor: 'grey_dark'
};

export default Banner;
