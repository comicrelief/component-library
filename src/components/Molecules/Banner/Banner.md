# Banner

``` js
import Text from '../../Atoms/Text/Text';

<Banner backgroundColor='purple_dark' >
  <Text tag="p" color='white'>
    Mental health issues affect 1 in 4 of us in the UK.
    Join now and with your help we can provide vital support for people like Jordan.
    You know what to do.
  </Text>
</Banner>
```
