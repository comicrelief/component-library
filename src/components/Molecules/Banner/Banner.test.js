import React from 'react';
import 'jest-styled-components';
import renderWithTheme from '../../../../tests/hoc/shallowWithTheme';
import Banner from './Banner';
import Text from '../../Atoms/Text/Text';

it('renders correctly', () => {
  const tree = renderWithTheme(
    <Banner backgroundColor="purple_dark">
      <Text tag="p" color="white">
        Mental health issues affect 1 in 4 of us in the UK.
        Join now and with your help we can provide vital support for people like Jordan.
        You know what to do.
      </Text>
    </Banner>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
