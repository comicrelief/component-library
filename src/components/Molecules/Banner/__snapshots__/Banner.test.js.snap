// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c2 {
  color: #FFFFFF;
  font-size: 1rem;
  line-height: 1rem;
  text-transform: inherit;
  line-height: normal;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

.c0 {
  background: #48276E;
}

.c1 {
  text-align: center;
  padding: calc(1.5rem * 2) 1.5rem;
  margin: 0 auto;
  max-width: 1440px;
}

@media (min-width:740px) {
  .c1 {
    padding: 4rem 2rem;
  }
}

@media (min-width:1440px) {
  .c1 {
    padding: 4rem 4rem;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <p
      className="c2"
      color="white"
      size="s"
    >
      Mental health issues affect 1 in 4 of us in the UK. Join now and with your help we can provide vital support for people like Jordan. You know what to do.
    </p>
  </div>
</div>
`;
